### PR TITLE
Update webpack

### PR DESCRIPTION
**Pull request by aDEVE**
Split and clean of webpack config. 

My PR is a:

💅 Enhancement

Main update on the:

Framework

### CODE REVIEW

File: packages/strapi-helper-plugin/lib/internals/webpack/configs/paths.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+/**
+ * rootAdminpath : Path of admin in main directory of application.
+ * For example : 
+ *   - build from monorepository : appPath = 'path/strapi/packages/strapi-admin'
+ *   - build in app dev : appPath = `path/${nameProject}/admin`
+ */
+const rootAdminpath =  __IS_MONOREPO__ ? strapiAdmin : appPathAdmin;
+
+const adminPath = __IS_ADMIN__ && __IS_MONOREPO__ ? strapiAdmin : path.resolve(__PWD__);
```
Comment (soupette) [inline packages/strapi-helper-plugin/lib/internals/webpack/configs/paths.js]: `adminPath` and `rootAdminPath` seems similar unless you're referring to a plugin's admin?
I think you need to add comments on that one.

File: packages/strapi-helper-plugin/lib/internals/webpack/configs/paths.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+const resolveApp = relativePath => path.resolve(appDirectory, relativePath);
+
+module.exports = {
+  adminPath: adminPath,
+  appPathAdmin: appPathAdmin,
+  appIndexJs: resolveApp('src/index.js'),
```
Comment (soupette) [inline packages/strapi-helper-plugin/lib/internals/webpack/configs/paths.js]: If `resolveApp` is used only once wouldn't it be better to write `appIndexJs: path.resolve(appDirectory, 'src/index.js')`?

File: packages/strapi-helper-plugin/lib/internals/webpack/configs/paths.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+ *   - build in app dev : appPath = `path/${nameProject}/admin`
+ */
+const rootAdminpath =  __IS_MONOREPO__ ? strapiAdmin : appPathAdmin;
+
+const adminPath = __IS_ADMIN__ && __IS_MONOREPO__ ? strapiAdmin : path.resolve(__PWD__);
+
+const resolveApp = relativePath => path.resolve(appDirectory, relativePath);
+
+module.exports = {
+  adminPath: adminPath,
```
Comment (soupette) [inline packages/strapi-helper-plugin/lib/internals/webpack/configs/paths.js]:
```
module.exports = {
  adminPath,
  appPath,
// ...
};
```

File: packages/strapi-helper-plugin/lib/internals/webpack/webpack.base.babel.js [hunk 4]
```diff
@@ -2,153 +2,52 @@
...
+const { COMMON_ALIAS } = require('./configs/alias');
+const foldersToInclude = require('./configs/foldersToInclude');
+const { __NODE_ENV__, __PROD__ } = require('./configs/globals');
+const paths = require('./configs/paths');
+const URLs = require('./configs/server');
+const pkg = require(paths.packageJson);
+const pluginId = pkg.name.replace(/^strapi-/i, '');
 
 module.exports = options => {
+  const { alias, babelPresets, devtool, disableExtractTextPlugin, entry, externals,  output, plugins } = options;
```
Comment (soupette) [inline packages/strapi-helper-plugin/lib/internals/webpack/webpack.base.babel.js]: I think the object destructuring makes the configuration harder to understand here.

File: packages/strapi-helper-plugin/lib/internals/webpack/webpack.base.babel.js [hunk 5]
```diff
@@ -2,153 +2,52 @@
...
+  const commonDevtool = 'cheap-module-source-map';
+
   return {
-    entry: options.entry,
+    entry: entry,
     output: Object.assign(
       {
         // Compile into js/build.js
-        path: path.join(adminPath, 'admin', 'build'),
+        path: path.join(paths.adminPath, 'admin', 'build'),
```
Comment (soupette) [inline packages/strapi-helper-plugin/lib/internals/webpack/webpack.base.babel.js]: You should put this logic un your `paths.js` file

File: packages/strapi-helper-plugin/lib/internals/webpack/webpack.dev.babel.js [hunk 6]
```diff
@@ -6,81 +6,22 @@ const fs = require('fs');
...
+const port = argv.port || __PORT__ || 3000;
 
 module.exports = require('./webpack.base.babel')({
   // Add hot reloading in development
   entry: Object.assign(
     {
       main: [
         `webpack-hot-middleware/client?path=http://localhost:${port}/__webpack_hmr`,
-        path.join(appPath, 'admin', 'admin', 'src', 'appDev.js'),
+        path.join(paths.appPath, 'admin', 'admin', 'src', 'appDev.js'),
```
Comment (soupette) [inline packages/strapi-helper-plugin/lib/internals/webpack/webpack.dev.babel.js]: Same here this path should be imported from your `paths.js`

File: packages/strapi-helper-plugin/lib/internals/webpack/webpack.dll.babel.js [hunk 7]
```diff
@@ -7,23 +7,25 @@
...
 const webpack = require('webpack');
+const { COMMON_ALIAS } = require('./configs/alias');
 const isAdmin = process.env.IS_ADMIN === 'true';
 
-const isSetup = process.env.IS_MONOREPO || false;
 const appPath = process.env.APP_PATH || path.resolve(process.env.PWD, '..', ( isAdmin ? '' : '..' ));
 
+
+const isSetup = process.env.IS_MONOREPO;
+
```
Comment (soupette) [inline packages/strapi-helper-plugin/lib/internals/webpack/webpack.dll.babel.js]: Why don't you use your `paths.js` to get the `rootAdminPath` ?

Comment (soupette) [reply packages/strapi-helper-plugin/lib/internals/webpack/webpack.dll.babel.js]: Is this one different?

File: packages/strapi-helper-plugin/lib/internals/webpack/webpack.dll.babel.js [hunk 8]
```diff
@@ -7,23 +7,25 @@
...
   }
 
   return path.resolve(appPath, 'admin');
 })();
-
 module.exports = {
   context: appPath,
   entry: {
-    vendor: ['react', 'react-dom', 'react-intl', 'reactstrap', 'react-transition-group', 'immutable', 'lodash', 'babel-polyfill'], // Shared dependencies accross the admin and plugins.
+    vendor: ['react-dnd', 'react-dnd-html5-backend', 'react', 'react-dom', 'react-intl', 'reactstrap', 'react-transition-group', 'immutable', 'lodash', 'babel-polyfill'], // Shared dependencies accross the admin and plugins.
```
Comment (soupette) [inline packages/strapi-helper-plugin/lib/internals/webpack/webpack.dll.babel.js]: Why don't you use the `dependencies.json` file you created above?

File: packages/strapi-helper-plugin/package.json [hunk 9]
```diff
@@ -105,5 +105,8 @@
     "webpack-dev-middleware": "^1.12.0",
     "webpack-hot-middleware": "^2.18.2",
     "whatwg-fetch": "^2.0.3"
+  },
+  "devDependencies": {
+    "webpack-cli": "^3.1.2"
```
Comment (soupette) [inline packages/strapi-helper-plugin/package.json]: Why did you add this dependency?

File: packages/strapi/lib/middlewares/router/index.js [hunk 10]
```diff
@@ -36,7 +36,7 @@ module.exports = strapi => {
           composeEndpoint(value, null, router)(cb);
         });
 
-        // router.prefix(strapi.config.admin.path || `/${strapi.config.paths.admin}`);
+        // router.prefix(strapi.config.admin.path || `/${strapi.config.paths.appPathAdmin}`);
```
Comment (soupette) [inline packages/strapi/lib/middlewares/router/index.js]: You should delete this line

File: packages/strapi-helper-plugin/lib/internals/webpack/configs/paths.js (hunk 1 above)
Comment (MartinMuzatko) [reply packages/strapi-helper-plugin/lib/internals/webpack/configs/paths.js]: Yes, I am also a bit confused on the differences. Is this change required in order to develop strapi as monorepo or as individual package?

File: packages/strapi-helper-plugin/lib/internals/webpack/webpack.base.babel.js (hunk 4 above)
Comment (MartinMuzatko) [reply packages/strapi-helper-plugin/lib/internals/webpack/webpack.base.babel.js]: If you have a lot of code, it is meaningful to keep the grouping/object. Agreed

**Comment by lauriejim** [comment]
@soupette The PR you link is merged. Do you want to make some tests before merging this PR ?

**Comment by soupette** [comment]
I don't think that this PR adds value to the existing codebase I rather close it.
